src/main/scala/io/iohk/ethereum/vm/WorldStateProxy.scala

```diff
@@ -25,8 +25,20 @@ trait WorldStateProxy[WS <: WorldStateProxy[WS, S], S <: Storage[S]] { self: WS
   protected def noEmptyAccounts: Boolean
   protected def accountStartNonce: UInt256 = UInt256.Zero
 
-  def combineTouchedAccounts(world: WS): WS
 
+  /*
+  *  During tx: 0xcf416c536ec1a19ed1fb89e4ec7ffb3cf73aa413b3aa9b77d60e4fd81a4296ba
+  *  precompiled account 0000000000000000000000000000000000000003 was deleted in block 2675119,
+  *  even though the deletion should have been reverted due to an out of gas error.
+  *  It was due to erroneous implementations of eip161 in parity and geth.
+  *  To avoid rewinding the chain, this special case is covered in parity and geth.
+  *  more details:
+  *  https://github.com/ethereum/EIPs/issues/716
+  *  https://github.com/ethereum/go-ethereum/pull/3341/files#r89548312
+  * */
+  def keepPrecompileTouched(world: WS): WS
+
+  protected val ripmdContractAddress = Address(3)
```

Comment:
what stands for `ripmd`?

Reply:
it is family of hashing functions, and contract under `Address(3)` returns ripmd hash of input